### PR TITLE
MGMT-8788: move all helper package to one package

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
Currently we have quite a few small utility packages:
hash,rbac, osversion,slice,warn,color,conditions.
Thi PR move them all under one package (utils). This package
has no need for defining a type interface for it

### CODE REVIEW

File: pkg/utils/slice.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func FindCRFile(a []*chart.File, x string) int {
 	return -1
 }
 
-func Insert(a []string, index int, value string) []string {
+func SliceInsert(a []string, index int, value string) []string {
```
Comment (qbarrand) [inline pkg/utils/slice.go]: How about `StringSliceInsert`?
in Go 1.18, which brings generics, we'll be able to make those functions work with any type 🕺🏻

File: pkg/utils/slice.go [hunk 2]
```diff
@@ -16,7 +16,7 @@ func Find(a []string, x string) int {
 }
 
 // Contains tells whether a contains x.
-func Contains(a []string, x string) bool {
+func SliceContains(a []string, x string) bool {
```
Comment (qbarrand) [inline pkg/utils/slice.go]: How about `StringSliceContains`?

File: pkg/utils/slice.go [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+package utils
 
 import (
 	"helm.sh/helm/v3/pkg/chart"
 )
 
 // Find returns the smallest index i at which x == a[i],
 // or len(a) if there is no such index.
-func Find(a []string, x string) int {
+func SliceFind(a []string, x string) int {
```
Comment (qbarrand) [inline pkg/utils/slice.go]: How about `StringSliceFind`?

File: pkg/utils/warn.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+// OnErrorOrNotFound warn on error or not found
+func OnErrorOrNotFound(found bool, err error) {
+	if !found || err != nil {
+		log.Info(Print("OnErrorOrNotFound: "+err.Error(), Brown))
+	}
+}
+
+// OnError warn on error
+func OnError(err error) {
```
Comment (qbarrand) [inline pkg/utils/warn.go]: How about `WarnOnError`?

File: pkg/utils/warn.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package utils
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var log = zap.New(zap.UseDevMode(true)).WithName(Print("warning", Brown))
+
+// OnErrorOrNotFound warn on error or not found
+func OnErrorOrNotFound(found bool, err error) {
```
Comment (qbarrand) [inline pkg/utils/warn.go]: How about `WarnOnErrorOrNotFound`?

File: pkg/utils/slice.go (hunk 3 above)
Comment (yevgeny-shnaidman) [reply pkg/utils/slice.go]: fixed

File: pkg/utils/slice.go (hunk 2 above)
Comment (yevgeny-shnaidman) [reply pkg/utils/slice.go]: fixed

File: pkg/utils/slice.go (hunk 1 above)
Comment (yevgeny-shnaidman) [reply pkg/utils/slice.go]: fixed

File: pkg/utils/warn.go (hunk 4 above)
Comment (yevgeny-shnaidman) [reply pkg/utils/warn.go]: fixed

File: pkg/utils/warn.go (hunk 5 above)
Comment (yevgeny-shnaidman) [reply pkg/utils/warn.go]: fixed

**Pull request merged**